File: python/XMLparser.py
import xml.etree.ElementTree as ET
from user import User

def read_users_from_xml(filename):
    # Parse the XML file
    tree = ET.parse(filename)
    root = tree.getroot()
    
    users = []
    for user_elem in root.findall('user'):
        user_id = int(user_elem.find('user_id').text)
        name = user_elem.find('name').text
        email = user_elem.find('email').text
        users.append(User(user_id, name, email))
    
    return users


users = read_users_from_xml("../files/file5.xml")

for user in users:
    print(user.printUser())

File: C#/XmlParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Xml.Linq;
using UserSpace;
namespace XmlParser
{
    class XmlUserParser
    {
        public static List<User> ReadUsersFromXml(string filename)
        {
            XDocument doc = XDocument.Load(filename);
            var users = doc.Descendants("user")
                           .Select(user => new User(
                               int.Parse(user.Element("user_id")?.Value),
                               user.Element("name").Value,
                               user.Element("email").Value))
                           .ToList();
            return users;
        }
    }
}